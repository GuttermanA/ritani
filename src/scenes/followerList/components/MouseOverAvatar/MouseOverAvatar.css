/* .follower-container {
  position: relative;
  width: 50%;
  padding: 1rem;
} */

/* .follower {
  position: relative;
} */
.button.absolute {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
}

div.avatar-container {
  position: relative;
  padding: 0.1rem;
  flex: 1 0 15%;
}

img.avatar {
  border-radius: 50%;
  max-width: 3.5rem;
  /* min-width: 25%; */
  /* height: auto; */
}

@media only screen and (min-height: 1080px) and (max-height: 1439px) {
  img.avatar {
    max-width: 8.5rem;
  }
}

@media only screen and (min-height: 1440px) {
  img.avatar {
    max-width: 11.5rem;
  }
}
