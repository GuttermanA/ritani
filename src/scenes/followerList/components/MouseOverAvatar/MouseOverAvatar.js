import React, { Component } from "react";
import { Image, Button, Container } from "components";
import "./MouseOverAvatar.css";

class MouseOverAvatar extends Component {
  state = { showButton: false, disabled: false };

  handleMouseEnter = event => {
    this.setState({ showButton: true });
  };

  handleMouseLeave = event => {
    this.setState({ showButton: false });
  };

  handleClick = async event => {
    event.preventDefault();
    event.stopPropagation();
    const { id, fetchUserWithFollowers } = this.props;
    this.setState({ disabled: true });
    await fetchUserWithFollowers(id);
    this.setState({ disabled: false });
  };

  render() {
    const {
      src,
      alt,
      id,
      url,
      followersURL,
      fetchUserWithFollowers,
      ...rest
    } = this.props;
    const { disabled } = this.props;
    return (
      <Container
        data-testid="avatar-container"
        className="avatar-container"
        onMouseEnter={this.handleMouseEnter}
        onMouseLeave={this.handleMouseLeave}
      >
        <Image className="avatar" src={src} alt={alt} {...rest} />
        {this.state.showButton && (
          <Button
            data-testid="follower-button"
            className="button absolute"
            onClick={this.handleClick}
            disabled={disabled}
          >
            {id}
          </Button>
        )}
      </Container>
    );
  }
}

export default MouseOverAvatar;
