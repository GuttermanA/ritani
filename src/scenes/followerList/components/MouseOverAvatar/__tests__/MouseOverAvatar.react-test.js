import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import MouseOverAvatar from "../MouseOverAvatar";

describe("<MouseOverAvatar/>", () => {
  afterEach(cleanup);
  it("renders correctly", () => {
    const component = render(<MouseOverAvatar />);
    expect(component).toMatchSnapshot();
  });

  it("renders a button when hovered", () => {
    let component = render(<MouseOverAvatar />);
    expect(component).toMatchSnapshot();

    let container = component.getByTestId("avatar-container");

    fireEvent.mouseEnter(container);
    component = render(<MouseOverAvatar />);
    expect(component.getByTestId("follower-button")).toBeTruthy();
    fireEvent.mouseLeave(container);
    component = render(<MouseOverAvatar />);
    expect(component.queryByTestId("follower-button")).toBeNull();
  });
});
