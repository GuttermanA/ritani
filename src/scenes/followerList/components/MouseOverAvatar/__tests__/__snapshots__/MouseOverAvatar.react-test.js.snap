// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MouseOverAvatar/> renders a button when hovered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="avatar-container"
        data-testid="avatar-container"
      >
        <img
          class="avatar"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="avatar-container"
      data-testid="avatar-container"
    >
      <img
        class="avatar"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<MouseOverAvatar/> renders a button when hovered 2`] = `
<div
  className="avatar-container"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    className="avatar"
  />
  <button
    className="button absolute"
    onClick={[Function]}
    type="submit"
  />
</div>
`;

exports[`<MouseOverAvatar/> renders a button when hovered 3`] = `
<div
  className="avatar-container"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    className="avatar"
  />
</div>
`;

exports[`<MouseOverAvatar/> renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="avatar-container"
        data-testid="avatar-container"
      >
        <img
          class="avatar"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="avatar-container"
      data-testid="avatar-container"
    >
      <img
        class="avatar"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
